adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -54,6 +95,145 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     replaceRdd(tFn(rdd))
   }
 
+  /**
+   * Pipes genomic data to a subprocess that runs in parallel using Spark.
+   *
+   * Files are substituted in to the command with a $x syntax. E.g., to invoke
+   * a command that uses the first file from the files Seq, use $0.
+   *
+   * @param cmd Command to run.
+   * @param files Files to make locally available to the commands being run.
+   * @param flankSize Number of bases to flank each command invocation by.
+   * @return Returns a new GenomicRDD of type Y.
+   *
+   * @tparam X The type of the record created by the piped command.
+   * @tparam Y A GenomicRDD containing X's.
+   */
+  def pipe[X, Y <: GenomicRDD[X, Y], V <: InFormatter[T, U, V]](cmd: String,
```

Comment:
Could you add a Map parameter with values to add to the [`ProcessBuilder.environment`](http://docs.oracle.com/javase/8/docs/api/java/lang/ProcessBuilder.html#environment--)?